gax/src/main/java/com/google/api/gax/batching/BatcherImpl.java

```diff
@@ -257,9 +261,20 @@ public void onFailure(Throwable throwable) {
   }
 
   private void onBatchCompletion() {
-    if (numOfOutstandingBatches.decrementAndGet() == 0) {
-      synchronized (flushLock) {
+    boolean shouldClose = false;
+
+    synchronized (flushLock) {
+      if (numOfOutstandingBatches.decrementAndGet() == 0) {
```

Comment:
Maybe use `<= 0` just in case as a safer option guaranteeing we will not jump over 0 and go into negative infinity?

Reply:
I think that would create weird behavior. I dont think I want waiters on the flushLock to be notified multiple times